packages/node-integration-tests/utils/run-tests.ts

```diff
@@ -0,0 +1,61 @@
+/* eslint-disable no-console */
+import childProcess from 'child_process';
+import os from 'os';
+
+const testPaths = childProcess.execSync('jest --listTests', { encoding: 'utf8' }).trim().split('\n');
+
+let testsSucceeded = true;
+const testAmount = testPaths.length;
+const fails: string[] = [];
+
+const threads = os.cpus().map(async (_, i) => {
+  let testPath = testPaths.pop();
+  while (testPath !== undefined) {
```

Comment:
If by `forEach` you mean iterating over the jobs - I didn't do that because a) I don't want to overload the host machine by instantly starting a bazillion of processes, b) I wanted to distribute the jobs among the workers evenly, i.e. by popping another job off the queue when they're done with their current one. I honestly wouldn't know how to do that with `forEach`.

Reply:
I had in my head to just split the tests by cpus beforehand and then sync iterate, but the producer - consumer pattern you have here is way more effective, I just didn't think it through.